recipe/reader.go

```diff
@@ -31,14 +31,14 @@ func NewReader(pathToConfig string) *Reader {
 }
 
 //  Read loads the list of recipes from a give file or directory path.
-func (r *Reader) Read(path string) (recipes []Recipe, err error) {
+func (r *Reader) Read(lg log.Logger, path string) (recipes []Recipe, err error) {
```

Comment:
instead of passing logger on read, we can pass it when initiating reader via `NewReader`